### ISSUE TITLE
jdownloader fails to install with .Error: Could not create the Java Virtual Machine.

### ISSUE THREAD

**Issue by tkothe** [Homebrew/homebrew-cask#55517]
#### General troubleshooting steps

- [x] I have retried my command with `--force` and the issue is still present.
- [x] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [x] I made doubly sure this is not a [checksum does not match](https://github.com/Homebrew/homebrew-cask/blob/master/doc/reporting_bugs/a_cask_fails_to_install.md#checksum-does-not-match-error) error.
- [x] I ran `brew update-reset && brew update` and retried my command.
- [x] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [x] I checked there are no [open issues](https://github.com/Homebrew/homebrew-cask/issues) for the same problem.
- [x] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).

#### Description of issue

Installation of JDownloader fails with the error shown below. There is not much more to say to this other than that I did run `brew cask install homebrew/cask-versions/java8` as required and that did complete successfully.

#### Command that failed

```
$ brew cask install jdownloader
```

#### Output of command with `--force --verbose --debug`

```
$ brew cask install jdownloader --force --verbose --debug
==> Cask::Installer#install
==> Printing caveats
==> Caveats
jdownloader requires Java 8. You can install it with

  brew cask install homebrew/cask-versions/java8

==> Cask::Installer#fetch
==> Satisfying dependencies
==> Downloading
==> Downloading http://installer.jdownloader.org/clean/JD2Setup.dmg
/usr/bin/curl -q --show-error --user-agent Mozilla/5.0\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10_12_3\)\ AppleWebKit/602.4.8\ \(KHTML,\ like\ Gecko\)\ Version/10.0.3\ Safari/602.4.8 --location --silent --head http://installer.jdownloader.org/clean/JD2Setup.dmg
/usr/bin/curl -q --show-error --user-agent Mozilla/5.0\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10_12_3\)\ AppleWebKit/602.4.8\ \(KHTML,\ like\ Gecko\)\ Version/10.0.3\ Safari/602.4.8 --fail --location --remote-time --continue-at 0 --output /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg.incomplete http://installer.jdownloader.org/clean/JD2Setup.dmg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 5454k  100 5454k    0     0  4257k      0  0:00:01  0:00:01 --:--:-- 4257k
==> Checking quarantine support
/usr/bin/xattr
/usr/bin/swift /usr/local/Homebrew/Library/Homebrew/cask/utils/quarantine.swift
==> Quarantine is available.
==> Verifying Gatekeeper status of /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg
/usr/bin/xattr -p com.apple.quarantine /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg
==> /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg is not quarantined
==> Quarantining /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg
/usr/bin/swift /usr/local/Homebrew/Library/Homebrew/cask/utils/quarantine.swift /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg http://installer.jdownloader.org/clean/JD2Setup.dmg http://jdownloader.org/
==> Downloaded to -> /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg
==> No SHA-256 checksum defined for Cask 'jdownloader', skipping verification.
==> Installing Cask jdownloader
==> Cask::Installer#stage
hdiutil imageinfo /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg
==> Extracting primary container
==> Using container class UnpackStrategy::Dmg for /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg
hdiutil attach -plist -nobrowse -readonly -noidme -mountrandom /var/folders/l8/y43mqkhd24xggvfss7x2m_gh0000gn/T/d20181129-25745-19ljz9v /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg
find . -print0
mkbom -s -i /var/folders/l8/y43mqkhd24xggvfss7x2m_gh0000gn/T/20181129-25745-1txahyn.list -- /var/folders/l8/y43mqkhd24xggvfss7x2m_gh0000gn/T/20181129-25745-1tkc63k.bom
ditto --bom /var/folders/l8/y43mqkhd24xggvfss7x2m_gh0000gn/T/20181129-25745-1tkc63k.bom -- /private/var/folders/l8/y43mqkhd24xggvfss7x2m_gh0000gn/T/d20181129-25745-19ljz9v/dmg.EhgRnD /var/folders/l8/y43mqkhd24xggvfss7x2m_gh0000gn/T/d20181129-25745-1umferv
diskutil eject /private/var/folders/l8/y43mqkhd24xggvfss7x2m_gh0000gn/T/d20181129-25745-19ljz9v/dmg.EhgRnD
cp -pR /var/folders/l8/y43mqkhd24xggvfss7x2m_gh0000gn/T/d20181129-25745-1umferv/JDownloader\ Installer.app/. /usr/local/Caskroom/jdownloader/latest/JDownloader\ Installer.app
chmod -Rf +w /var/folders/l8/y43mqkhd24xggvfss7x2m_gh0000gn/T/d20181129-25745-1umferv
==> Verifying Gatekeeper status of /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg
/usr/bin/xattr -p com.apple.quarantine /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg
==> /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg is quarantined
==> Propagating quarantine from /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg to /usr/local/Caskroom/jdownloader/latest
/usr/bin/xattr -p com.apple.quarantine /Users/tillkothe/Library/Caches/Homebrew/downloads/53eb50893fe352f66dded615b83802398af02cdf4d252f562dbbffc36970768d--JD2Setup.dmg
/usr/bin/xargs -0 -- /bin/chmod -h u\+w
/usr/bin/xargs -0 -- /usr/bin/xattr -w -s com.apple.quarantine 0181\;5bfff986\;Homebrew\\x20Cask\;E378DE93-CEEA-49FF-93A3-24FDF18C3FD0
==> Creating metadata directory /usr/local/Caskroom/jdownloader/.metadata/latest/20181129143656.140.
==> Creating metadata subdirectory /usr/local/Caskroom/jdownloader/.metadata/latest/20181129143656.140/Casks.
==> Installing artifacts
==> 2 artifact/s defined
#<SortedSet:0x00007ff22bc907c8>
==> Installing artifact of class Cask::Artifact::PreflightBlock
/usr/local/Caskroom/jdownloader/latest/JDownloader\ Installer.app/Contents/MacOS/JavaApplicationStub -dir /Applications -q -Dinstall4j.suppressStdout=true -Dinstall4j.debug=false -VcreateDesktopLinkAction\$Boolean=false -VaddToDockAction\$Boolean=false
==> Purging files for version latest of Cask jdownloader
Error: Failure while executing; `/usr/local/Caskroom/jdownloader/latest/JDownloader\ Installer.app/Contents/MacOS/JavaApplicationStub -dir /Applications -q -Dinstall4j.suppressStdout=true -Dinstall4j.debug=false -VcreateDesktopLinkAction\$Boolean=false -VaddToDockAction\$Boolean=false` exited with 1. Here's the output:
<JAVA_HOME>/lib/ext exists, extensions mechanism no longer supported; Use -classpath instead.
.Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.


Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
/usr/local/Homebrew/Library/Homebrew/system_command.rb:107:in `assert_success'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:50:in `run!'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:27:in `run'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:31:in `run!'
/usr/local/Homebrew/Library/Homebrew/cask/dsl/base.rb:14:in `system_command'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/jdownloader.rb:11:in `block (2 levels) in load'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/abstract_flight_block.rb:39:in `instance_eval'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/abstract_flight_block.rb:39:in `abstract_phase'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/abstract_flight_block.rb:22:in `install_phase'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:209:in `block in install_artifacts'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/set.rb:674:in `each'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/set.rb:674:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:200:in `install_artifacts'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:95:in `install'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/install.rb:21:in `block in run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/install.rb:14:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/install.rb:14:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/abstract_command.rb:34:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:89:in `run_command'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:155:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:120:in `run'
/usr/local/Homebrew/Library/Homebrew/cmd/cask.rb:7:in `cask'
/usr/local/Homebrew/Library/Homebrew/brew.rb:89:in `<main>'
Error: Kernel.exit
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:166:in `exit'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:166:in `rescue in run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:143:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:120:in `run'
/usr/local/Homebrew/Library/Homebrew/cmd/cask.rb:7:in `cask'
/usr/local/Homebrew/Library/Homebrew/brew.rb:89:in `<main>'

```

#### Output of `brew cask doctor`

```
$ brew cask doctor
==> Homebrew Version
1.8.4-9-g30c90dc
==> macOS
10.13.6
==> SIP
Enabled
==> Java
11.0.1, 11, 10.0.2, 1.8.0_192, 1.8.0_144
==> Gatekeeper support
Enabled
==> Homebrew Cask Install Location
<NONE>
==> Homebrew Cask Staging Location
/usr/local/Caskroom
==> Homebrew Cask Taps:
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask (4070 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-versions (209 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-eid (11 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-fonts (1185 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-drivers (158 casks)
==> Contents of $LOAD_PATH
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/ruby-macho-2.1.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/rubocop-rspec-1.30.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/rubocop-0.60.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/unicode-display_width-1.4.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/ruby-progressbar-1.10.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/rainbow-3.0.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/powerpack-0.1.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/plist-3.4.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/parser-2.5.3.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/parallel-1.12.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/jaro_winkler-1.5.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/extensions/universal-darwin-18/2.3.0/jaro_winkler-1.5.1
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/backports-3.11.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/ast-2.4.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/activesupport-5.2.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/tzinfo-1.2.5/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/thread_safe-0.3.6/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/minitest-5.11.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/i18n-1.1.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/concurrent-ruby-1.0.5/lib
/Library/Ruby/Gems/2.3.0/gems/did_you_mean-1.0.0/lib
/Library/Ruby/Site/2.3.0
/Library/Ruby/Site/2.3.0/x86_64-darwin17
/Library/Ruby/Site/2.3.0/universal-darwin17
/Library/Ruby/Site
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0/x86_64-darwin17
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0/universal-darwin17
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/x86_64-darwin17
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/universal-darwin17
/usr/local/Homebrew/Library/Homebrew
==> Environment Variables
LC_ALL="en_US.UTF-8"
PATH="/usr/local/Homebrew/Library/Homebrew/shims/scm:/usr/bin:/bin:/usr/sbin:/sbin"
SHELL="/bin/bash"
```

#### Output of `brew tap`

```
$ brew tap
browsh-org/browsh
buo/cask-upgrade
heroku/brew
homebrew/boneyard
homebrew/cask
homebrew/cask-drivers
homebrew/cask-eid
homebrew/cask-fonts
homebrew/cask-versions
homebrew/core
homebrew/dupes
homebrew/php
homebrew/science
homebrew/versions
homebrew/x11
vitorgalvao/tiny-scripts
```

**Comment by vitorgalvao** [comment]
My VM has gone in the crapper, so I can’t test this. Two questions:

1. Do you see a GUI at any point in the installation process?
2. Is it still happening to you?

I’ll also reference I’m considering closing this, as it’s a relatively popular cask and there’s been no other reports.

**Comment by tkothe** [comment]
Hey @vitorgalvao, thanks so much for getting back regarding this. I tried a couple more things and can confirm this is not an issue with the `jdownloader` cask. I successfully installed the cask on a different machine (which is running Mojave).

Also, **I found that the `java8` cask seems to be the culprit causing the trouble though.** Even though it is listed as a requirement for the `jdownloader` cask, I was able to install the `jdownloader` cask without having `java8` cask installed on the Mojave machine. 
On my High Sierra machine though, `java8` seemed to be installed, but was failing on `remove` and `reinstall`. Though after I ran `brew cask reinstall java8` (which failed on the installation part of `java8`), I was able to successfully install `jdownloader`. I also see that I have `jenv` installed on both machines and on the High Sierra machine it seems to be broken...

Long story short: I am closing this issue. Most probably it was just my locally misconfigured Java. Sorry for the trouble!
